### DISCUSSION TITLE
Problem with using API for add set via postman

### DISCUSSION THREAD

**Discussion by evorotnikov** [smartin015/continuousprint#205] [Q&A] [answered]
Hi there, i'm trying to play with the API a bit and figure out if it possible to create a set and add it to the LAN queue via API. Overall i have couple of questions/problems:

1. I checked the source code and looks like add set will add a set to "DEFAULT_QUEUE". Does that mean it will add it to default local queue? or is there a way to explicitly say to which queue i want set to be added.
2. I'm trying to hit get status from local postman and getting 403 back unfortunately, however it's working pretty good via browser. Is there any setup that i need to do to be able to trigger those requests i would say externally from OctoPrint/ContiniousePrint perspective? 

Thanks a lot in advance!

**Comment by smartin015** [comment, marked as answer]
Hi @evorotnikov!

1. You're correct, `/set/add` will always add your set to the local queue. That's mostly for legacy reasons, and could be amended to include a target queue in the request if you want to file a feature request :)
2. CPQ uses OctoPrint's access control under the hood for its endpoints - if you're querying via the browser, the API/session key is already included which authenticates your request. Check out [one of the examples](https://github.com/smartin015/continuousprint/blob/master/api_examples/example.py#L5) to see how to set an API key to get access to CPQ's API.

Let me know if you hit any friction - happy to help!

**Reply by evorotnikov** [reply]
Thanks a lot Scott, let me create a feature request then!
